### ISSUE TITLE
beforeSave method is called multiple times

### ISSUE THREAD

**Issue by illmat** [quietcreep/meteor-formation#27]
While browsing your code, I've recognized that you're calling the beforeSave method once in the saveInstance function and once inside the Formation.insert method. Is this on purpose?

**Comment by quietcreep** [comment]
Yes, this is on purpose. It's so that the beforeSave hook can contain ```Meteor.isServer``` and ```Meteor.isClient``` checks. If it has some side effects from being called twice, it might be wise to use those checks to make sure it does the appropriate thing on either side.

Is it causing you trouble, and do you have an idea how to circumvent it?
